### ISSUE TITLE
Custom callback lua condition text nil error 2.0.7

### ISSUE THREAD

**Issue by ceylina** [Sonaza/Candy#9]
so out of curiosity, I tested an example callback you have as an example

> local durability = tonumber(strmatch(text,"(%d+)")) or 0; return durability <= 50;

It throws this lua error trying to validate it. 
At first I thought the callback might be bad but I have an existing lua condition that never threw an error when clicking validate prior to 2.0.6/2.0.7

> return UnitLevel("player") == GetMaxPlayerLevel();

```lua
Candy\options-Options.lua:36: attempt to call method 'GetModuleText' (a nil value)
Candy\options-Options.lua:36: in function <Candy\options.lua:21>
Candy\options-Options.lua:73: in function `OnCancel'
FrameXML\StaticPopup.lua:4428: in function `StaticPopup_OnClick'
[string "*:OnClick"]:1: in function <[string "*:OnClick"]:1>

Locals:
script = "local durability = tonumber(strmatch(text,"(%d+)")) or 0; return durability <= 80;"
broker = "gmDurability"
script = "return (function(text, icon) local durability = tonumber(strmatch(text,"(%d+)")) or 0; return durability <= 80; end)(...)"
compiled = <function> defined Candy-Visibility-gmDurability:0
scriptError = nil
_ = CandygmDurabilityFrame {
 0 = <userdata>
 isVisible = true
 fadein = <unnamed> {
 }
 data = <table> {
 }
 fadeout = <unnamed> {
 }
 broker = "gmDurability"
 background = CandygmDurabilityFrameBackground {
 }
 text = CandygmDurabilityFrameText {
 }
}
module = <table> {
}
(*temporary) = nil
(*temporary) = <table> {
 SetDefaultModuleLibraries = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:398
 IsPlayerInAGroup = <function> defined @Candy\core.lua:195
 EnableModule = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:363
 modules = <table> {
 }
 baseName = "Candy"
 IterateEmbeds = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:473
 UnlockBars = <function> defined @Candy\core.lua:619
 RemoveCandy = <function> defined @Candy\core.lua:371
 MODIFIER_STATE_CHANGED = <function> defined @Candy\core.lua:173
 GetFrameKeyPoints = <function> defined @Candy\core.lua:833
 PLAYER_REGEN_DISABLED = <function> defined @Candy\core.lua:164
 CompileVisibilityScript = <function> defined @Candy\core.lua:235
 GetActiveBrokers = <function> defined @Candy\options.lua:395
 IsEnabled = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:482
 PlayerInInstance = <function> defined @Candy\core.lua:185
 DisableModule = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:381
 UpdateVisibility = <function> defined @Candy\core.lua:250
 RegisterMessage = <function> defined @AngryAssignments\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:116
 UnregisterMessage = <function> defined @AngryAssignments\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:179
 StripColor = <function> defined @Candy\core.lua:208
 db = <table> {
 }
 SetEnabledState = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:455
 enabledState = true
 RegisterEvent = <function> defined @AngryAssignments\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:116
 DataObjectCreated = <function> defined @Candy\core.lua:433
 IterateModules = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:468
 ContextMenu = CandyMenuFrame {
 }
 PrepareTooltip = <function> defined @Candy\core.lua:754
 PET_BATTLE_OVER = <function> defined @Candy\core.lua:181
 OnInitialize = <function> defined @Candy\core.lua:49
 NewModule = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:266
 GetModule = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:241
 PET_BATTLE_OPENING_START = <function> defined @Candy\core.lua:177
 defaultModuleLibraries = <table> {
 }
 CloseOptions = <function> defined @Candy\options.lua:414
 UpdateCandyBars = <function> defined @Candy\core.lua:199
 ActiveBars = <table> {
 }
 OpenCandyOptions = <function> defined @Candy\options.lua:449
 ChangeJustify = <function> defined @Candy\core.lua:637
 GetName = <function> defined @AngryAssignments\libs\AceAddon-3.0\AceAddon-3.0.lua:310
 SetGlobalFrameStrata = <function> defined @Candy\options.lua:439
 FrameHasParent = <function> defined @Candy\core.lua:862
 ConvertAnchor = <function> defined @Candy\core.lua:647
 name = "Candy"
 SetGlobalFontOu
```

**Comment by Sonaza** [comment]
This was a bug left in the 2.0.7 version by accident and I already fixed it for version 2.0.8.

**Comment by ceylina** [comment]
Thanks for the fast fix!
